### ISSUE TITLE
Разработка внешней компоненты для использования альтернативной криптографии

### ISSUE THREAD

**Issue by SergSignalCOM** [1C-Company/DirectBank#16]
Требуется заменить используемую в DirectBank криптографию(на базе криптопровайдера) на альтернативную (на базе библиотеки Message-PRO). Для этого, я так понимаю, требуется разработать свою внешнюю компоненту.

В документации для вариант с внешней компонентой - https://github.com/1C-Company/DirectBank/blob/master/doc/vk-transport-api/readme.md 
приведен перечень функций, которые должна реализовывать внешняя компонента:

1. Функции общего назначения 
2. Работа с криптографией, ключами ЭП, хранилищами ключей ЭП 
3. Аутентификация на сервере банка 
4. Взаимодействие с сервером банка
 

Имеется ли возможность реализовать только функции связанные с криптографией, а остальной функционал использовать от типового варианта решения DirectBank, или такого варианта нет и все надо написать самому, включая транспортный уровень, аутентификацию на сервере, взаимодействие с сервером банка и т.п.?

**Comment by GlebovAndrei** [comment]
Да, все правильно понимаете - внешняя компонента должна заменить все функции.
Поддерживать еще один альтернативный вариант (только криптография для DirectBank во внешней компоненте) не планируем.
Что мешает реализовать продукт на базе Message-PRO, который будет поддерживать вызовы CryptoAPI? Тогда это приведет к типовому варианту использования DirectBank...

**Comment by SergSignalCOM** [comment]
Можно уточнить? 
"Реализовать продукт на базе Message-PRO, который будет поддерживать вызовы CryptoAPI "
Это имеется ввиду реализовать MS CSP(криптопровайдер) или реализовать MS CryptoAPI(модуль -  альтернативный стандартной Microsoft библиотеки)?

**Comment by SergSignalCOM** [comment]
Какую библиотеку CryptoApi вы используете под операционными системами Linux и Mac OC.

**Comment by GlebovAndrei** [comment]
> Это имеется в виду реализовать MS CSP(криптопровайдер) или реализовать MS CryptoAPI(модуль - альтернативный стандартной Microsoft библиотеки)?

**Comment by GlebovAndrei** [comment]
"Какую библиотеку CryptoApi вы используете под операционными системами Linux и Mac OC?"
===
Уточним у разработчиков платформы и ответим

**Comment by kisttan** [comment]
можно ли вызывать внешнее СКЗИ минуя MS CryptoApi, через вызов COM или JavaScript?

**Comment by SevrD** [comment]
Можно, но только для варианта с подключением внешнего модуля

**Comment by SevDmitry** [comment]
"Какую библиотеку CryptoApi вы используете под операционными системами Linux и Mac OC?"
=== Платформа заявляет о поддержке CryptoPro на Linux.

**Comment by vovche** [comment]
> "Какую библиотеку CryptoApi вы используете под операционными системами Linux и Mac OC?"
> === Платформа заявляет о поддержке CryptoPro на Linux.
Подскажите, VipNet CSP платформа до сих пор под Linux не поддерживает?

**Comment by SevDmitry** [comment]
VipNet CSP для Linux пока не вышел (только бета)
https://infotecs.ru/downloads/all/vipnet-csp.html?arrFilter_93=1824101684&set_filter=Y
Платформа о поддержке бета-версий никогда не заявляет.
По факту же может и поддерживаться без доработок.
Т.е. нужно сделать установку и указать пути к библиотекам.
